Pcap++/header/PcapFilter.h

```diff
@@ -484,82 +484,117 @@ namespace pcpp
 
 
 	/**
-	 * @class AndFilter
-	 * A class for connecting several filters into one filter with logical "and" between them. For example: if the 2 filters are: "IPv4 address =
-	 * x.x.x.x" + "TCP port dst = 80", then the new filter will be: "IPv4 address = x.x.x.x _AND_ TCP port dst = 80"<BR>
-	 * This class follows the composite design pattern<BR>
+	 * @class CompositeFilter
+	 * The base class for all filter classes composed of several other filters. This class is virtual and abstract, hence cannot be instantiated.<BR>
 	 * For deeper understanding of the filter concept please refer to PcapFilter.h
-	 * @todo add some methods: "addFilter", "removeFilter", "clearAllFilter"
 	 */
-	class AndFilter : public GeneralFilter
+	class CompositeFilter : public GeneralFilter
 	{
-	private:
+	protected:
 		std::vector<GeneralFilter*> m_FilterList;
 	public:
-
 		/**
-		 * An empty constructor for this class. Use addFilter() to add filters to the and condition
+		 * An empty constructor for this class. Use addFilter() to add filters to the composite filter.
 		 */
-		AndFilter() {}
+		CompositeFilter() {};
 
 		/**
-		 * A constructor that gets a list of pointers to filters and creates one filter from all filters with logical "and" between them
+		 * A constructor that gets a list of pointers to filters and creates one filter from all filters
 		 * @param[in] filters The list of pointers to filters
 		 */
-		explicit AndFilter(std::vector<GeneralFilter*>& filters);
+		explicit CompositeFilter(const std::vector<GeneralFilter*>& filters);
```

Comment:
Although `CompositeFilter` is abstract and cannot be instantiated, I'd still move the c'tors to be `protected` instead of `public` to be even more explicit

Reply:
The change of the ctors from `public` to `protected` would require `CompositeLogicFilter` to have its ctors be explicitly redeclared as `public` instead of the current `using CompositeFilter::CompositeFilter;` as the ctors declared with a using declaration retain their access level from the base class.

Reply:
ok, I agree 👍